### DISCUSSION TITLE
C++ example on Windows

### DISCUSSION THREAD

**Discussion by quartetwest** [snakers4/silero-vad#522] [Q&A] [answered]
First time posting here. I just started working with silero-vad.

I'm trying to get the .cpp example running on Windows. I'm usually developing on Linux, but the project I'm working on requires VAD in a Windows C++ program. This is for the x64 architecture. 

I have downloaded and successfully built the ONNX RunTime from the latest source code (from Microsoft repo).

I have also tried using the prebuilt ONNXRuntime packages (1.18.1, 1.18.0, 1.12.1) for Windows, but I'm observing the same issues.  

I believe I'm using the latest silero_vad.onnx model file, which does seem to load. The code throws an exception when calling predict(r)

Here's the snippet of relevant code
	std::vector<float> input_wav(...)   //this vector gets loaded correctly with input wave file data 

    	std::wstring path = L"silero_vad.onnx";
    	VadIterator vad(path); // This seems to load fine - I've walked through the code in the debugger

    	vad.process(input_wav, output_wav); // this throws an exception when calling predict(r). See code snippet below. The error is "Invalid input name: state"

void process(const std::vector<float>& input_wav)
{
    reset_states();

    audio_length_samples = input_wav.size();

    for (int j = 0; j < audio_length_samples; j += window_size_samples)
    {
        if (j + window_size_samples > audio_length_samples)
            break;
        std::vector<float> r{ &input_wav[0] + j, &input_wav[0] + j + window_size_samples };
        predict(r); // this ultimately calls ort_outputs = session->Run(...) below, and this is where the exception is thrown
    }
}

void predict(const std::vector<float> &data)
{
    // Infer
    // Create ort tensors
    input.assign(data.begin(), data.end());
    Ort::Value input_ort = Ort::Value::CreateTensor<float>(
        memory_info, input.data(), input.size(), input_node_dims, 2);
    Ort::Value state_ort = Ort::Value::CreateTensor<float>(
        memory_info, _state.data(), _state.size(), state_node_dims, 3);
    Ort::Value sr_ort = Ort::Value::CreateTensor<int64_t>(
        memory_info, sr.data(), sr.size(), sr_node_dims, 1);

    // Clear and add inputs
    ort_inputs.clear();
    ort_inputs.emplace_back(std::move(input_ort));
    ort_inputs.emplace_back(std::move(state_ort));
    ort_inputs.emplace_back(std::move(sr_ort));

    // Infer - this is where the exception is thrown 
    ort_outputs = session->Run(
        Ort::RunOptions{nullptr},
        input_node_names.data(), ort_inputs.data(), ort_inputs.size(),
        output_node_names.data(), output_node_names.size());

I would appreciate any help or guidance on next steps. Please let me know if I can provide any additional data. 

My goal is run SileroVAD and compare to a couple other solutions that I'm evaluating.

**Comment by quartetwest** [comment, marked as answer]
Just wanted to provide an update here. I was able to get this working (Windows, C++, ONNX), and my testing thus far with silero-vad is proving to be quite good. Impressive speed and accuracy. I'm observing the same PR curve as depicted in the repo.

**Reply by snakers4** [reply]
If you would like to share the example, if the Windows one differs, please do
